content/300-accelerate/650-troubleshoot.mdx

```diff
@@ -0,0 +1,124 @@
+---
+title: 'Troubleshooting Guide'
+metaTitle: 'Troubleshooting: Prisma Accelerate'
+metaDescription: 'Troubleshooting Prisma Accelerate.'
+tocDepth: 3
+toc: true
+---
+
+<TopBlock>
+
+When working with Prisma Accelerate, you may encounter errors often highlighted by specific error codes during development and operations. It is important to understand the meaning of these errors, why they occur, and how to resolve them in order to ensure the smooth operation of your applications. This guide aims to provide insights and steps to troubleshoot specific error codes encountered with Prisma Accelerate.
+
+</TopBlock>
+
+## <inlinecode>[P6009](/orm/reference/error-reference#p6009-responsesizelimitexceeded)</inlinecode> (<inlinecode>ResponseSizeLimitExceeded</inlinecode>)
+
+This error is triggered when the response size from a database query exceeds the 5MB limit. We've implemented this restriction to safeguard your application performance, as retrieving data over 5MB can significantly slow down your application due to multiple network layers. Typically, transmitting more than 5MB of data is common when conducting ETL (Extract, Transform, Load) operations. However, for other scenarios such as transactional queries, real-time data fetching for user interfaces, bulk data updates, or aggregating large datasets for analytics outside of ETL contexts, it should generally be avoided. These use cases, while essential, can often be optimised to work within the 5MB limit, ensuring smoother performance and a better user experience.
+
+### Possible causes for <inlinecode>[P6009](/orm/reference/error-reference#p6009-responsesizelimitexceeded)</inlinecode>
+
+#### Transmitting images/files in response
+
+This error may arise if images or files stored in base64 format within your table are being fetched, resulting in a large response size. Storing assets in base64 format is generally discouraged because it significantly impacts database performance and scalability. In addition to performance, it makes database backups slow and significantly increases the cost of storing routine backups.
+
+**Suggested solution:** Store the image/file in a BLOB store like Cloudflare [R2](https://developers.cloudflare.com/r2/), [AWS S3](https://aws.amazon.com/pm/serv-s3/), [Cloudinary](https://cloudinary.com/), etc. These services would allow you to store the assets optimally and return a URL through which you should be able to access the asset. You can store this URL in your table instead of storing the asset in a base64 format. This would substantially reduce the size of your response.
+
+#### Incorrect/Missing <inlinecode>[where](/orm/reference/prisma-client-reference#where)</inlinecode> or <inlinecode>[select](/orm/reference/prisma-client-reference#select)</inlinecode> clause
```

Comment:
Everything that is said makes sense. It might be more readable to break out the where and select into separate paragraphs. So one for for `missing select` and one for `incorrect where`. It reads a bit confusing since this block talks about two issues at once. If you want to group those together I would add a headline a la `Overfetching of data`.